src/elements/content-sidebar/__tests__/BoxAISidebar.test.tsx

```diff
@@ -1,55 +1,172 @@
 import React from 'react';
-import {userEvent} from '@testing-library/user-event';
-import {render, screen} from '../../../test-utils/testing-library';
-import BoxAISidebarComponent, {BoxAISidebarProps} from '../BoxAISidebar';
+import { userEvent } from '@testing-library/user-event';
+import { screen, render } from '../../../test-utils/testing-library';
+import BoxAISidebar, { BoxAISidebarProps } from '../BoxAISidebar';
 
-const mockOnClearClick = jest.fn();
+jest.mock('@box/box-ai-agent-selector', () => ({
+    ...jest.requireActual('@box/box-ai-agent-selector'),
+    BoxAiAgentSelectorWithApi: () => <div data-testid="sidebar-agent-selector" />
+}));
+
+const mockOnClearAction = jest.fn();
+jest.mock('@box/box-ai-content-answers', () => ({
+    ...jest.requireActual('@box/box-ai-content-answers'),
+    withApiWrapper: Component => props => (<Component 
+        createSession={props.createSessionRequest}
+        encodedSession={props.restoredSession}
+        error={null}
+        getAIStudioAgents={props.getAIStudioAgents}
+        hostAppName={props.hostAppName}
+        hasCustomSuggestedQuestions={false}
+        isAgentSelectorEnabled={props.isAgentSelectorEnabled}
+        isAIStudioAgentSelectorEnabled={props.isAIStudioAgentSelectorEnabled}
+        isCitationsEnabled={props.isCitationsEnabled}
+        isDebugModeEnabled={props.isDebugModeEnabled}
+        isMarkdownEnabled={props.isMarkdownEnabled}
+        isLoading={false}
+        isOpen
+        isResetChatEnabled={props.isResetChatEnabled}
+        isStreamingEnabled={props.isStreamingEnabled}
+        itemID={props.itemID}
+        itemIDs={props.itemIDs}
+        onClearAction={mockOnClearAction}
+        onCloseModal={jest.fn()}
+        onSelectAgent={jest.fn()}
+        onAgentEditorToggle={jest.fn()}
+        questions={props.restoredQuestions}
+        retryQuestion={jest.fn()}
+        sendQuestion={jest.fn()}
+        shouldRenderProviders={jest.fn()}
+        stopQuestion={jest.fn()}
+        suggestedQuestionsRequestState="success"
+        suggestedQuestions={props.suggestedQuestions}
+        warningNotice={props.warningNotice}
+        warningNoticeAriaLabel={props.warningNoticeAriaLabel}
```

Comment:
can we use the spread operator here to pass in the props and set all the props int the mockProps?

Reply:
We could with some of them, but I think that could be a bit misleading. This wrapper is not passing all props, so if we would just do spread operator we would change how it behaves outside of mock (that's why in BoxAISidebar we are using context to pass values to BoxAISidebarContent). That could lead to code that is passing in UTs, but not reality.